### PR TITLE
Added bounds/bbox to Graphhopper Query

### PR DESCRIPTION
**Pull request by jjjakobsen**

### CODE REVIEW

File: src/Provider/GraphHopper/GraphHopper.php [hunk 1]
```diff
@@ -71,6 +76,11 @@ public function geocodeQuery(GeocodeQuery $query): Collection
 
         $url = sprintf(self::GEOCODE_ENDPOINT_URL, urlencode($address), $this->apiKey, $query->getLocale(), $query->getLimit());
 
+        $bounds = $query->getBounds();
+        if ($bounds) {
+            $url .= sprintf(self::GEOCODE_BOUNDS_PARAMS, $bounds->getWest(), $bounds->getSouth(), $bounds->getEast(), $bounds->getNorth());
```
Comment (jbelien) [inline src/Provider/GraphHopper/GraphHopper.php:81]:
```suggestion
            $url .= sprintf('&bbox=%f,%f,%f,%f', $bounds->getWest(), $bounds->getSouth(), $bounds->getEast(), $bounds->getNorth());
```

`GEOCODE_BOUNDS_PARAMS` can be removed.

File: src/Provider/GraphHopper/Tests/GraphHopperTest.php [hunk 2]
```diff
@@ -62,8 +63,11 @@ public function testGeocodeWithRealAddressAndLocale()
         }
 
         $provider = new GraphHopper($this->getHttpClient($_SERVER['GRAPHHOPPER_API_KEY']), $_SERVER['GRAPHHOPPER_API_KEY']);
-        $results = $provider->geocodeQuery(GeocodeQuery::create('242 Acklam Road, London, United Kingdom')->withLocale('fr'));
-
+        $results = $provider->geocodeQuery(
+            GeocodeQuery::create('242 Acklam Road, London, United Kingdom')
+                ->withLocale('fr')
+                ->withBounds(new Bounds(50, -10, 55, 10))
+        );
```
Comment (jbelien) [inline src/Provider/GraphHopper/Tests/GraphHopperTest.php:70]: Please use a new test to test this new "feature".

**Review by jbelien — CHANGES_REQUESTED** [review]
Thanks for your contribution! 👍 

Tests are currently failing (see https://github.com/geocoder-php/Geocoder/runs/5153173637?check_suite_focus=true) ;
You need to also commit the `cached_responses` by running the tests locally (with a valid key).

**Comment by jbelien** [comment]
No worries, it's quite simple.
You just need to run the tests locally and the cached responses will be automatically generated and stored in `Tests/.cached_responses` folder. Then, you just have to commit those files and push it.

To run the tests locally:

1. Go to `src/Provider/GraphHopper` folder
2. Install dependencies : `composer install`
3. Set your GraphHopper API key in `phpunit.xml.dist`
4. Run the tests : `composer run test`
5. Cached responses will be generated in `src/Provider/GraphHopper/Tests/.cached_responses`
6. Commit the cached response files (**do NOT commit `phpunit.xml.dist`**)
7. Push!

---

The [`cache.md`](https://github.com/geocoder-php/Geocoder/blob/master/docs/cookbook/cache.md) is about our (old) Cache Plugin, now refactor to [Cache Provider](https://github.com/geocoder-php/cache-provider) ; it's unrelated to the tests.

**Comment by jjjakobsen** [comment]
Thanks for the hint, yes that was easy.

I just pushed an update, hope all is good now

Am Fr., 11. Feb. 2022 um 15:31 Uhr schrieb Jonathan Beliën <
***@***.***>:

File: src/Provider/GraphHopper/GraphHopper.php (hunk 1 above)
Comment (jjjakobsen) [reply src/Provider/GraphHopper/GraphHopper.php]: done

File: src/Provider/GraphHopper/Tests/GraphHopperTest.php (hunk 2 above)
Comment (jjjakobsen) [reply src/Provider/GraphHopper/Tests/GraphHopperTest.php]: done

File: src/Provider/GraphHopper/GraphHopper.php [hunk 3]
```diff
@@ -71,6 +71,11 @@ public function geocodeQuery(GeocodeQuery $query): Collection
 
         $url = sprintf(self::GEOCODE_ENDPOINT_URL, urlencode($address), $this->apiKey, $query->getLocale(), $query->getLimit());
 
+        $bounds = $query->getBounds();
+        if ($bounds) {
```
Comment (jbelien) [inline src/Provider/GraphHopper/GraphHopper.php:75]: This check needs to be more strict!

**Review by jbelien — CHANGES_REQUESTED** [review]

File: src/Provider/GraphHopper/GraphHopper.php (hunk 3 above)
Comment (jjjakobsen) [reply src/Provider/GraphHopper/GraphHopper.php]: done

**Review by jbelien — APPROVED** [review]

**Comment by jbelien** [comment]
Thanks! 👍 
A new release has been published: https://packagist.org/packages/geocoder-php/graphhopper-provider

**Pull request merged**